File: app/src/main/java/com/porkandlager/travelonbudget/wires/retrofitter/ApiService.java
package com.porkandlager.travelonbudget.wires.retrofitter;

import com.porkandlager.travelonbudget.wires.models.acta.Acta;
import com.porkandlager.travelonbudget.wires.models.requests.FlightSearchRequestMeta;
import com.porkandlager.travelonbudget.wires.models.responses.FlightSearchWithBudgetResponse;

import retrofit2.Retrofit;
import rx.Observable;

/**
 * Created by tista on 11/12/16.
 */

public enum ApiService {
    INSTANCE;

    private TravelOnBudgetService travelOnBudgetService;

    public void setRetrofit(Retrofit retrofit) {
        this.travelOnBudgetService = retrofit.create(TravelOnBudgetService.class);
    }

    public Observable<FlightSearchWithBudgetResponse> searchFlights(Acta<FlightSearchRequestMeta> acta) {
        return travelOnBudgetService.getFlights(acta);
    }

    public static ApiService getInstance() {
        return INSTANCE;
    }

}

File: app/src/main/java/com/porkandlager/travelonbudget/wires/retrofitter/ApiService.kt
package com.porkandlager.travelonbudget.wires.retrofitter

import com.porkandlager.travelonbudget.wires.models.acta.Acta
import com.porkandlager.travelonbudget.wires.models.requests.FlightDetailImageRequestMeta
import com.porkandlager.travelonbudget.wires.models.requests.FlightSearchRequestMeta
import com.porkandlager.travelonbudget.wires.models.responses.FlightDetailImageSearchResponse
import com.porkandlager.travelonbudget.wires.models.responses.FlightSearchWithBudgetResponse

import retrofit2.Retrofit
import rx.Observable

/**
 * Created by tista on 11/12/16.
 */

enum class ApiService {
    instance;

    private var travelOnBudgetService: TravelOnBudgetService? = null

    fun setRetrofit(retrofit: Retrofit) {
        this.travelOnBudgetService = retrofit.create(TravelOnBudgetService::class.java)
    }

    fun searchFlights(acta: Acta<FlightSearchRequestMeta>): Observable<FlightSearchWithBudgetResponse> {
        return travelOnBudgetService!!.getFlights(acta)
    }

    fun searchPhotos(acta: Acta<FlightDetailImageRequestMeta>): Observable<FlightDetailImageSearchResponse> {
        return travelOnBudgetService!!.getPhotosFromAirport(acta)
    }

}
